server/middleware/checkPrisonerInCaseloadMiddleware.test.ts

```diff
@@ -0,0 +1,353 @@
+import createError from 'http-errors'
+import { Request, Response } from 'express'
+import type { SessionData } from 'express-session'
+import aValidPrisonerSummary from '../testsupport/prisonerSummaryTestDataBuilder'
+import checkPrisonerInCaseload from './checkPrisonerInCaseloadMiddleware'
+
+describe('checkPrisonerInCaseloadMiddleware', () => {
```

Comment:
Whilst HMPPS Prisoner Profile has a test for `checkPrisonerInCaseloadMiddleware`, I found it difficult to read and port into our codebase.
In the end I decided to rewrite the test from scratch to cover all scenarios / code paths.

This meant I had to fully understand all of the code paths and functionality of `checkPrisonerInCaseloadMiddleware` which is a good thing. If nothing else it means I can explain the logic that we have copied from Prisoner Profile in this respect 👍